File: mlbusinesscomponents/src/main/java/com/mercadolibre/android/mlbusinesscomponents/components/pickup/SizeType.java
package com.mercadolibre.android.mlbusinesscomponents.components.pickup;

import com.mercadolibre.android.mlbusinesscomponents.R;

public enum SizeType {
    SMALL(R.dimen.ui_fontsize_xsmall, R.dimen.ui_1_75m),
    MEDIUM(R.dimen.ui_fontsize_small, R.dimen.ui_2m);

    private final int fontSize;
    private final int imageSize;

    SizeType(final int fontSize, final int imageSize) {
        this.fontSize = fontSize;
        this.imageSize = imageSize;
    }

    public int getFontSize() {
        return fontSize;
    }

    public int getImageSize() {
        return imageSize;
    }

}

File: mlbusinesscomponents/src/main/java/com/mercadolibre/android/mlbusinesscomponents/components/pickup/SizeType.kt
package com.mercadolibre.android.mlbusinesscomponents.components.pickup

import com.mercadolibre.android.mlbusinesscomponents.R

enum class SizeType(val fontSize: Int, val imageSize: Int) {
    SMALL(R.dimen.ui_fontsize_xsmall, R.dimen.ui_2m),
    MEDIUM(R.dimen.ui_fontsize_small, R.dimen.ui_image_size_medium);

    companion object {
        @JvmStatic
        fun getFontSizeOrDefault(size: String?, defaultSize: Int) : Int {
            return size
                ?.let { values().firstOrNull { value -> value.name.equals(it, ignoreCase = true) } }
                ?.fontSize
                ?: defaultSize
        }

        @JvmStatic
        fun getImageSizeOrDefault(size: String?, defaultSize: Int) : Int {
            return size
                ?.let { values().firstOrNull { value -> value.name.equals(it, ignoreCase = true) } }
                ?.imageSize
                ?: defaultSize
        }
    }
}